tests/test_pool.py

```diff
@@ -0,0 +1,222 @@
+import datetime
+import unittest
+
+from unittest import IsolatedAsyncioTestCase
+from unittest.mock import Mock
+from pyfreenas import Machine
+
+from pyfreenas.pool import Pool, PoolStatus
+from tests.fakes.fakeserver import (
+    FreeNASServer,
+    TDiskQueryResult,
+    TDiskTemperaturesResult,
+    TPoolQueryResult,
+    TVmQueryResult,
+)
+from typing import (
+    Any,
+    Dict,
+    List,
+    Union,
+)
+
+
+class TestPool(IsolatedAsyncioTestCase):
+    _server: FreeNASServer
+    _machine: Machine
+
+    def setUp(self):
+        self._server = FreeNASServer()
+        self._server.register_method_handler(
+            "disk.query", lambda *args: [],
+        )
+        self._server.register_method_handler(
+            "vm.query", lambda *args: [],
+        )
+
+    async def asyncSetUp(self):
+        self._machine = await Machine.create(
+            self._server.host,
+            username=self._server.username,
+            password=self._server.password,
+            secure=False,
+        )
+
+    async def asyncTearDown(self):
+        await self._machine.close()
+        await self._server.stop()
+
+    async def test_pool_data_interpretation(self) -> None:
+        ENCRYPT = 0
+        GUID = "1234ABCD"
+        ID = 100
+        IS_DECRYPTED = True
+        NAME = "testpool"
+        self._server.register_method_handler(
+            "pool.query",
+            lambda *args: [
+                {
+                    "encrypt": ENCRYPT,
+                    "encryptkey": "",
+                    "guid": GUID,
+                    "id": ID,
+                    "is_decrypted": IS_DECRYPTED,
+                    "name": NAME,
+                    "scan": {
```

Comment:
Should I leave this out, since it's not being tested below? I guess I was thinking about creating authentic data, but since it's not tested, there really is not purpose.

Reply:
I'm okay dropping scan from the query right now.  It looks like it is used for scrubs, and we can always come back to implement that later.

I don't think either one of us needs it right now, so let's focus on a minimum viable product.  We only need pools because datasets need pools, so let's implement what you need for your use case now :)

Reply:
OK. Removed _scrub_ from query and tests.